### PR TITLE
Fix flatten function for nested objects

### PR DESCRIPTION
**Pull request by mansishr**
`flatten_json.flatten_preserve_lists()` isn't smashing hierarchies beyond third nested level. The fix is to use `flatten_json.flatten()` instead and preserve lists while unflatting using `flatten_json.unflatten_list()`

### CODE REVIEW

**Comment by mansishr** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by psfoley** [comment]
Thank you for fixing this, @mansishr! I think once some tests for this are in place, this PR is ready to merge.

File: openfl/native/native.py [hunk 1]
```diff
@@ -70,11 +70,15 @@ def flatten(config, return_complete=False):
...
     Update the plan with the provided override and save it to disk.
+    If the override value is a list, keys corresponding to each list
+    value should be specified.
 
     For a list of available override options, call `fx.get_plan()`
 
     Args:
-        override_config : dict {"COMPONENT.settings.variable" : value}
+        override_config : dict {"COMPONENT.settings.variable" : value,
+                                "COMPONENT.settings.list_variable.0" : list_value_0,
```
Comment (psfoley) [inline openfl/native/native.py]: @mansishr does this mean that list items must be set individually? Or can a full list be provided here? i.e.

```
"COMPONENT.settings.list_variable" : [1,2,3]
```

Comment (mansishr) [reply openfl/native/native.py]: List items would need to be set individually for this to work

`{"COMPONENT.settings.list_variable.0" : 1, "COMPONENT.settings.list_variable.1" : 2, "COMPONENT.settings.list_variable.2" : 3}`

Comment (psfoley) [reply openfl/native/native.py]: This is not ideal for long lists. I would recommend adding some logic in the update plan method to do this transformation programmatically:

```
final_override_config = {}
for key, val in override_config.items():
    if type(val) == list:
        for idx,v in enumerate(val):
            final_override_config[f'{key}.{idx}'] = v
    else:
        final_override_config[key] = val
```

**Comment by psfoley** [comment]
LGTM!

**Comment by psfoley** [comment]
LGTM!

**Pull request merged**